legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/AppendOnlyPlanner.java

```diff
@@ -115,7 +116,13 @@ else if (!ingestMode().dataSplitField().isPresent())
     @Override
     public LogicalPlan buildLogicalPlanForPreActions(Resources resources)
     {
-        return LogicalPlan.builder().addOps(Create.of(true, mainDataset())).build();
+        List<Operation> operations = new ArrayList<>();
+        operations.add(Create.of(true, mainDataset()));
+        if (options().createStagingDataset())
+        {
+            operations.add(Create.of(true, stagingDataset()));
+        }
+        return LogicalPlan.of(operations);
```

Comment:
This seems to be duplicated...  Wonder if there is some opportunities to make this default operations, and let other implementations add subsequent operations.

Maybe like `options().defaultOperations();`